hyperspy/axes.py

```diff
@@ -95,14 +124,44 @@ def __init__(self,
         self.axes_manager = None
         self.on_trait_change(self.update_axis,
                              ['scale', 'offset', 'size'])
-        self.on_trait_change(self.update_value, 'index')
-        self.on_trait_change(self.set_index_from_value, 'value')
         self.on_trait_change(self._update_slice, 'navigate')
         self.on_trait_change(self.update_index_bounds, 'size')
         # The slice must be updated even if the default value did not
         # change to correctly set its value.
         self._update_slice(self.navigate)
 
+    def _index_changed(self, name, old, new):
+        self.events.index_changed.trigger(obj=self, index=self.index)
+        if not self._suppress_update_value:
+            new_value = self.axis[self.index]
+            if new_value != self.value:
+                self.value = new_value
+
+    def _value_changed(self, name, old, new):
+        old_index = self.index
+        new_index = self.value2index(new)
+        if self.continuous_value is False:  # Only values in the grid alowed
+            if old_index != new_index:
+                self.index = new_index
+                if new == self.axis[self.index]:
+                    self.events.value_changed.trigger(obj=self, value=new)
+            elif old_index == new_index:
+                new_value = self.index2value(new_index)
+                if new_value == old:
+                    self._suppress_value_changed_trigger = True
+                    self.value = new_value
```

Comment:
Where is `self._suppress_value_changed_trigger` restored? If you forgot it, then please add, if the state is supposed to be fixed in the new call to `_value_changed`, then the assumed code flow should be outlined in comments.